### PR TITLE
Initial implementation for Refocus notification

### PR DESCRIPTION
**Pull request by yjanezou**
@rev @sundeepsf

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+					} catch (Exception e) {
+						_logger.error("Failure - send Refocus sample'{}'. Exception '{}'", refocusSample.toJSON(), e);
+					} finally {
+						if(post != null){
+							post.releaseConnection();
+						}
+					}
+				}
+			} catch (RuntimeException ex) {
+				throw new SystemException("Failed to send an GOC++ notification.", ex);
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java]: not GOC++

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schedule/DistributedDatabaseSchedulingService.java [hunk 2]
```diff
@@ -494,7 +494,7 @@ public void run() {
 				} catch(Exception e) {
 					_logger.error("Exception occured when computing alert evaluation kpi metric - "+ ExceptionUtils.getFullStackTrace(e));
 				} finally {
-					_logger.error("marking alert with id {} for deletion", alert.getId().intValue());
+					_logger.error("marking alert with name {} and id {} for deletion", alert.getName(), alert.getId() == null? null: alert.getId().intValue());
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schedule/DistributedDatabaseSchedulingService.java]: when would alert id be null?

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	}
+
+	@Override
+	protected void clearAdditionalNotification(NotificationContext context) {
+		_sendRefocusNotification(context, false);
+	}
+
+	private void _sendRefocusNotification(NotificationContext context, boolean isTriggerActive) {
+		List<String> aspectPaths = context.getNotification().getSubscriptions();
+		String token = context.getNotification().getCustomText();
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java]: this is not really a good idea, like we discussed

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schedule/DistributedDatabaseSchedulingService.java (hunk 2 above)
Comment (yjanezou) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/service/schedule/DistributedDatabaseSchedulingService.java]: it happened to me in local testing, though later I figured out there was a bug in some other place, here the NPE complicated the troubleshooting process.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java (hunk 3 above)
Comment (yjanezou) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/RefocusNotifier.java]: now it is using configged value

**Pull request merged**